tests/cli/serve/serve_test.py

```diff
@@ -36,3 +37,233 @@ def testOstorlabServe_whenRefreshApiKeyFlagIsNotProvided_getOrCreateApiKey(
 
     assert mocked_refresh_api_key.called is False
     assert "API key refreshed" not in result.output
+
+
+def testOstorlabServe_whenStarting_shouldPersistPredifinedAgentGroups(
```

Comment:
Please split this to have one for each agent group

Reply:
it doesnt make any sense to do that !